modules/serviceregistry/src/main/java/org/opencastproject/serviceregistry/impl/JobDispatcher.java

```diff
@@ -0,0 +1,646 @@
+  private TrustedHttpClient client;
+
+  /** The thread pool to use for dispatching. */
+  protected ScheduledThreadPoolExecutor scheduledExecutor = null;
+
+  /** The factory used to generate the entity manager */
+  private EntityManagerFactory emf = null;
+
+  private ScheduledFuture jdfuture = null;
+
+  /**
+   * A list with job types that cannot be dispatched in each interation
+   */
+  private List<String> undispatchableJobTypes = null;
+
+  /** The dispatcher priority list */
+  protected final Map<Long, String> dispatchPriorityList = new HashMap<>();
+
+  /** OSGi DI */
+  @Reference(target = "(osgi.unit.name=org.opencastproject.common)")
+  void setEntityManagerFactory(EntityManagerFactory emf) {
+    this.emf = emf;
+  }
+
+  /** OSGi DI */
+  @Reference()
+  void setServiceRegistry(ServiceRegistryJpaImpl sr) {
+    this.serviceRegistry = sr;
+  }
+
+  @Reference()
+  void setOrganizationDirectoryService(OrganizationDirectoryService organizationDirectoryService) {
+    this.organizationDirectoryService = organizationDirectoryService;
+  }
+
+  @Reference
+  void setUserDirectoryService(UserDirectoryService svc) {
+    this.userDirectoryService = svc;
+  }
+
+  @Reference
+  void setSecurityService(SecurityService sec) {
+    this.securityService = sec;
+  }
+
+  @Reference
+  void setTrustedHttpClient(TrustedHttpClient client) {
+    this.client = client;
+  }
+
+  @Activate
+  public void activate(ComponentContext cc) throws ConfigurationException  {
+    logger.info("Activate job dispatcher");
+    scheduledExecutor = (ScheduledThreadPoolExecutor) Executors.newScheduledThreadPool(1);
+    scheduledExecutor.setRemoveOnCancelPolicy(true);
+    logger.info("Activated");
+    updated(cc.getProperties());
+  }
+
+
+  @Modified
+  public void modified(ComponentContext cc) throws ConfigurationException {
+    logger.debug("Modified in job dispatcher");
+    updated(cc.getProperties());
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see org.osgi.service.cm.ManagedService#updated(java.util.Dictionary)
+   */
+  @SuppressWarnings("rawtypes")
+  public void updated(Dictionary properties) {
+
+    logger.info("Updating job dipatcher properties");
+
+    // NB: Read the javadoc for Boolean.valueOf.  This basically defaults to false.
+    boolean enableDispatch = Boolean.valueOf(StringUtils.trimToNull((String) properties.get(OPT_DISPATCHENABLED)));
+
+    long dispatchInterval = DEFAULT_DISPATCH_INTERVAL;
+    String dispatchIntervalString = StringUtils.trimToNull((String) properties.get(OPT_DISPATCHINTERVAL));
+    if (StringUtils.isNotBlank(dispatchIntervalString)) {
+      try {
+        dispatchInterval = Long.parseLong(dispatchIntervalString);
+      } catch (Exception e) {
+        logger.warn("Dispatch interval '{}' is malformed, setting to {}", dispatchIntervalString, MIN_DISPATCH_INTERVAL);
+        dispatchInterval = MIN_DISPATCH_INTERVAL;
+      }
+      if (dispatchInterval == 0) {
+        logger.info("Dispatching disabled");
```

Comment:
Functionally, nothing.  I thought it was clearer having both flags since you might want to template out the interval across a cluster, but only enable dispatch on one node.  I'm fine removing the boolean and just relying on the interval if that would be better.